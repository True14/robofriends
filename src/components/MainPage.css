@font-face {
  font-display: swap;
  font-family: "SEGA LOGO FONT";
  font-style: normal;
  font-weight: normal;
  src: local("SEGA LOGO FONT"), url("../containers/SEGA.woff") format("woff");
}

h1 {
  font-family: "SEGA LOGO FONT";
  font-weight: 200;
  color: #0ccac4;
}

/* width */
::-webkit-scrollbar {
  width: 13px;
}

/* Track */
::-webkit-scrollbar-track {
  border-radius: 10px;
}

/* Handle */
::-webkit-scrollbar-thumb {
  background: white;
  border-radius: 10px;
}
