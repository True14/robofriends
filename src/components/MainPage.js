import React, { Component } from "react";
import CardList from "../components/CardList";
import SearchBox from "../components/SearchBox";
import Scroll from "../components/Scroll";
import Header from "../components/Header";
import ErrorBoundry from "../components/ErrorBoundry";
import "./MainPage.css";

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    return this.props.robots.filter((robot) => {
      return robot.name
        .toLowerCase()
        .includes(this.props.searchField.toLowerCase());
    });
  };

  render() {
    const { onSearchChange, isPending } = this.props;

    return (
      <div className="tc">
        <Header />
        <SearchBox id={"searchField"} searchChange={onSearchChange} />
        <Scroll>
          {isPending ? (
            <h1 className="tc">Loading</h1>
          ) : (
            <ErrorBoundry>
              <CardList robots={this.filterRobots()} />
            </ErrorBoundry>
          )}
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
