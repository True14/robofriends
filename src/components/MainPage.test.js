import { shallow } from "enzyme";
import React from "react";
import MainPage from "./MainPage";

describe("Test MainPage Component", () => {
  let wrapper;
  beforeEach(() => {
    const mockProps = {
      searchField: "sample text",
      onSearchChange: jest.fn(),
      onRequestRobots: jest.fn(),
      robots: [],
      isPending: false
    };

    wrapper = shallow(<MainPage {...mockProps} />);
  });

  it("should render the main page without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the main page without crashing when isPending is true", () => {
    const isPendingTrue = {
      searchField: "sample text",
      onSearchChange: jest.fn(),
      onRequestRobots: jest.fn(),
      robots: [],
      isPending: true
    };
    const isPendingTrueWrapper = shallow(<MainPage {...isPendingTrue} />);
    expect(isPendingTrueWrapper).toMatchSnapshot();
  });

  it("filters robots correctly", () => {
    const mockProps2 = {
      searchField: "john",
      onSearchChange: jest.fn(),
      onRequestRobots: jest.fn(),
      robots: [
        {
          id: 3,
          name: "John",
          email: "john@gmail.com"
        }
      ],
      isPending: false
    };
    const wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(wrapper2.instance().filterRobots()).toEqual([
      {
        id: 3,
        name: "John",
        email: "john@gmail.com"
      }
    ]);
  });

  it("filters robots correctly 2", () => {
    const mockProps3 = {
      searchField: "a",
      onSearchChange: jest.fn(),
      onRequestRobots: jest.fn(),
      robots: [
        {
          id: 3,
          name: "John",
          email: "john@gmail.com"
        }
      ],
      isPending: false
    };
    const wrapper3 = shallow(<MainPage {...mockProps3} />);
    expect(wrapper3.instance().filterRobots()).toEqual([]);
  });
});
