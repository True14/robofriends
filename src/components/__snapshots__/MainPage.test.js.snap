// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MainPage Component should render the main page without crashing 1`] = `
<div
  className="tc"
>
  <Memo(Header) />
  <Memo(SearchBox)
    id="searchField"
    searchChange={[MockFunction]}
  />
  <Scroll>
    <ErrorBoundry>
      <Memo(CardList)
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;

exports[`Test MainPage Component should render the main page without crashing when isPending is true 1`] = `
<div
  className="tc"
>
  <Memo(Header) />
  <Memo(SearchBox)
    id="searchField"
    searchChange={[MockFunction]}
  />
  <Scroll>
    <h1
      className="tc"
    >
      Loading
    </h1>
  </Scroll>
</div>
`;
